docs/faq/events.md

```diff
@@ -28,3 +30,182 @@ There are lots of times when words just aren't enough to really get clarity on a
+
+### Inside of your elements
+
+When adding event listening inside of your components, assuming you are using `lit-html` (or any other declarative renderer with event binding syntax) use the tools supplied therein to [add event listeners](https://lit-html.polymer-project.org/guide/writing-templates#add-event-listeners). The reasoning behind this is that for any listener added by `lit-html`, the same listener will be removed by `lit-html` when no longer needed; no ghost listeners! That means whether the DOM and its associated lister is available for the entire lifecycle of your application, or just for a small part of that, the least amount of work will be done to ensure the event is handled appropriately. See the `click` listener in the following code. It _only_ exits while `this.open` is `true`, which leaves the browser listening to less things and your code less likely to trigger handlers errantly.
+
+```js
+render() {
+    return html`
+        ${this.open
+            ? html`<button @click=${this.handleClose}>Close</button>
+            : html``
+        }
+    `;
+}
+```
+
+A side benefit of relying on the `@event` syntax in `LitElement` (if you use it) is that all events handled therein are automatically bound to `this` which will prevent the need to do so manually: no `this.handleClose = this.handleClose.bind(this);` and, no `handleClose = (event) => {}` in the case above.
+
+### On your elements
+
+If you are adding a listener to your element, from inside your element, you should add that listener in the `constructor`. The browser does a great job of garbage collecting that listener when the element is removed from the DOM, so there no reason to do any work to manage that yourself. See the following example where the element listeners for a `done` event.
+
+```js
+constructor() {
+    super();
+    this.addEventListener('done', this.handleDone);
+}
+handleDone = (event) => {
+    this.done = true;
+    // Do the work you'd like to complete when "done" here.
+}
+```
+
+Side note: as you will be adding this listener manually, you _will_ need to ensure that `this` is bound to your liking. See the class field method syntax above.
+
+### On elements outside of your element
+
+In some cases you will need to add event listener to elements outside of the element you are building: e.g. a `resize` listener of the `window`. In this case the `window` could last well beyond the last referenc to your element and it is very important that you do the work to ensure that the listener is added and removed appropriately. Thankfully, the custom elements specification outlines the [`connectedCallback` and `disconnectedCallback` lifecycle methods](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_custom_elements#Using_the_lifecycle_callbacks). These are the perfect hooks for managing these sorts of listeners. When adding event listeners to this sort of external element, use the `connectedCallback`, like so:
```

Comment:
```suggestion
In some cases you will need to add event listeners to elements outside of the element you are building: e.g. a `resize` listener on the `window`. In this case the `window` could last well beyond the last reference to your element and it is very important that you do the work to ensure that the listener is added and removed appropriately. Thankfully, the custom elements specification outlines the [`connectedCallback` and `disconnectedCallback` lifecycle methods](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_custom_elements#Using_the_lifecycle_callbacks). These are the perfect hooks for managing these sorts of listeners. When adding event listeners to this sort of external element, use the `connectedCallback`, like so:
```

Reply:
Changes?

Reply:
listener to listeners
referenc to reference